Add a link to Add-ons in the Settings

felipeerias opened the pull request:
Add a button to open the Add-ons page from the Settings, since that is a pretty obvious place to find this functionality.

To avoid messing up the layout of the honeycomb buttons, I've moved the "What's New" button closer to the version details. After some experimentation, I've placed two small buttons on the right side of the Settings window, linking to the Feedback and News pages.

The background of the What's New button becomes highlighted when the details of the new version are available (this replaces the little blue dot that we had before).

I have also simplified the implementation of the hidden shortcut to show the version details. Now it is just a matter of clicking on the app logo or the app name.

Finally, I have also simplified the layout in the Settings window by removing an unneeded LinearLayout and using the parent ConstraintLayout instead.

felipeerias commented:
This is how the dialog looks when there is new "What's New" information:

![Screenshot_com igalia wolvic dev_2024 12 06-20 53 17 419_016](https://github.com/user-attachments/assets/24d284ba-6199-4a59-81e8-c5c67d935514)

And this is how it looks in the regular state.

![Screenshot_com igalia wolvic dev_2024 12 06-20 51 32 799_385](https://github.com/user-attachments/assets/d05e5c72-7584-4ee5-9e1b-d020ce2611da)

I think that placing the Add-ons link here makes a lot of sense but am not completely convinced about the rest of the layout. What do you think?

svillar commented:
> I think that placing the Add-ons link here makes a lot of sense but am not completely convinced about the rest of the layout. What do you think?

I wholeheartedly agree with placing the addons there, now that I see it, it makes total sense to me. Regarding the other two elements I am not convinced either, looks too texty to me (the other elements are big icons)

felipeerias commented:
I have updated this PR to adjust the visual appearance.

This is how the Settings dialog looks when the "What's new" page for the current version has not been visited yet:

![Screenshot_com igalia wolvic dev_2024 12 17-23 58 00 520_505](https://github.com/user-attachments/assets/ecf67a21-69d3-45a5-a552-f394bc489d6d)

And this is how it looks otherwise:

![Screenshot_com igalia wolvic dev_2024 12 17-23 54 37 373_157](https://github.com/user-attachments/assets/1e5da5c5-78c9-4a4d-b5b9-5ff23c2fde2e)

felipeerias commented:
More information about the color attributes for the buttons is available at:

https://github.com/material-components/material-components-android/blob/1.9.0/docs/components/Button.md

svillar reviewed (changes requested):
I was about to approve it but I found a significant usability issue. If you open the addons from the settings then there is no clear way to exit the addons tab to go back to the normal browsing.

When visiting bookmarks or downloads is different because you see that the button is pressed in the navigation bar (that could be improved with colors too), but in the case of the addons there is no easy and intuitive way to go back. Somehow you have to discover that clicking on the bookmarks button in the nav bar can take you back

felipeerias commented:
@svillar Good catch. You can exit the Add-ons by clicking on the **✕** button at the top-right corner, but that is not very discoverable.

I think that there is also a bug in the code that updates the status of tray buttons. The Bookmarks button should become visibly active and clicking it should take you back to the browser. However, that code might not be working correctly in all cases.

felipeerias commented:
@svillar I have updated this PR so when you open the Add-ons panel from the Settings, in addition to the **✕** button on the top-right corner, the Bookmarks button in the Tray will also become active. Clicking that Bookmarks button will switch to showing the bookmarks, and clicking again will go back to Web browsing.

I have added a commit ensuring that the state of the Bookmarks and Downloads buttons in the Tray remains consistent through the different events.

svillar reviewed (approved):
lgtm
